pkg/entities/quest.go

```diff
@@ -166,6 +208,22 @@ func (e *Entity) UpdateUserQuestProgress(log *model.QuestUserLog) error {
 	if log.Action == model.BONUS || log.UserID == "" {
 		return nil
 	}
+
+	//validate quest action
+	err_action := e.validateQuestAction(string(log.Action))
+	// validate user
+	err_user := e.validateUser(log.UserID)
+
+	if err_action != nil && err_user == nil {
+		return err_action
+	}
+	if err_action == nil && err_user != nil {
+		return err_user
+	}
+	if err_action != nil && err_user != nil {
+		return errors.New(err_action.Error() + ", " + err_user.Error())
+	}
+
```

Comment:
why need to have 3 cases here?
```suggestion
	if err := e.validateQuestAction(string(log.Action)); err != nil {
		return err
	}
	if err := e.validateUser(log.UserID); err != nil {
		return err
	}
```